projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts

```diff
@@ -1043,7 +1043,7 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
         if (this.getDimensionType(dimension) === PivotDimensionType.Row) {
             const relatedDims = PivotUtil.flatten([dimension]).map(x => x.memberName);
             const content = this.rowDimensionContentCollection.filter(x => relatedDims.indexOf(x.dimension.memberName) !== -1);
-            const headers = flatten(content.map(x => x.headerGroups.toArray())).map(x => x.header.refInstance);
+            const headers = content.map(x => x.headerGroups.toArray()).flat().map(x => x.header && x.header.refInstance);
```

Comment:
By default this flattens only 1 level deep when a parameter is not passed. Do you always have just a single level here?

Reply:
@rkaraivanov We've debugged it and discussed it in the team. It is only 1 level we need to flatten.